### PR TITLE
Default master width

### PR DESCRIPTION
**Pull request by marcusramberg**
Allow setting default width in percent for master. Can be set per tag.

### CODE REVIEW

**Comment by avahe-kellenberger** [comment]
This is great, but I have two thoughts:

- Could we have it be per-tag? And also keep the monitor default
- We should have checks so the percentage must be > 0 and < 100. We could just clamp the value between 1 and 99, then log a warning if the given value was outside that range

**Comment by marcusramberg** [comment]
@avahe-kellenberger I've addressed your comments here as well as fixed config reloading to respect this setting now, so I guess we're ready for another review.

**Comment by avahe-kellenberger** [comment]
I'll check it out by tomorrow, thanks!

File: src/nimdowpkg/config/tagsettings.nim [hunk 1]
```diff
@@ -31,19 +32,30 @@ proc parseTagSetting(tagSetting: var TagSetting, settingsTable: TomlTableRef) =
       raise newException(Exception, "Invalid numMasterWindows for tag")
     tagSetting.numMasterWindows = numMasterWindows.intVal.int
 
+  if settingsTable.hasKey("defaultMasterWidthPercentage"):
+    let masterWidthSetting = settingsTable["defaultMasterWidthPercentage"]
+    if masterWidthSetting.kind == TomlValueKind.Int:
+      tagSetting.defaultMasterWidthPercentage = masterWidthSetting.intVal.int.clamp(10,90)
```
Comment (avahe-kellenberger) [inline src/nimdowpkg/config/tagsettings.nim:38]: I think here we should check if they provided a value outside of the 10-90 range, as well. 
Something like:
```nim
let givenPercent = masterWidthSetting.intVal
if givenPercent < 10 or givenPercent > 90:
  log "Invalid defaultMasterWidthPercentage, clamped to 10-90%", lvlWarn
tagSetting.defaultMasterWidthPercentage = givenPercent.clamp(10, 90)
```

**Review by avahe-kellenberger — APPROVED** [review]
Looks good, worked great for me

**Pull request merged**